lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/openturns/FunctionalChaosValidation.hxx

```diff
@@ -0,0 +1,102 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The validation of a chaos expansion
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_FUNCTIONALCHAOSVALIDATION_HXX
+#define OPENTURNS_FUNCTIONALCHAOSVALIDATION_HXX
+
+#include "openturns/Sample.hxx"
+#include "openturns/FunctionalChaosResult.hxx"
+#include "openturns/MetaModelValidation.hxx"
+#include "openturns/SplitterImplementation.hxx"
+#include "openturns/LeaveOneOutSplitter.hxx"
+#include "openturns/KFoldSplitter.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class FunctionalChaosValidation
+ *
+ * The validation of a chaos expansion
+ */
+
+class OT_API FunctionalChaosValidation
+  : public MetaModelValidation
+{
+  CLASSNAME
+
+public:
+  typedef Collection<Function>           FunctionCollection;
+
+  /** Default constructor */
+  FunctionalChaosValidation();
+
+  /** Parameter constructor */
+  FunctionalChaosValidation(const FunctionalChaosResult & chaosResult);
+
+  /** Parameter constructor */
+  FunctionalChaosValidation(const FunctionalChaosResult & chaosResult,
+                            const LeaveOneOutSplitter & splitter);
+
+  /** Parameter constructor */
+  FunctionalChaosValidation(const FunctionalChaosResult & chaosResult,
+                            const KFoldSplitter & splitter);
```

Comment:
Maybe we can have the following ctor instead : 
```cxx
FunctionalChaosValidation(const FunctionalChaosResult & chaosResult,
                          const KFoldSplitter & splitter,
                          const UnsignedInteger kFold = ResourceMap::GetAsUnsignedInteger("FunctionalChaosValidation-DefaultKFoldParameter");
```

Reply:
I does not seem necessary to me, since KFoldSplitter can manage a default value for K, isn't it?

Reply:
It allows user to set the value directly without "playing" with the `ResourceMap`

Reply:
Consider for example the unit test in `t_LinearModelValidation.py` where we use K-Fold C.V.:
```
# Create KFold validation
splitterKFold = ot.KFoldSplitter(sampleSize, kFoldParameter)
validationKFold = otexp.LinearModelValidation(
    result, splitterKFold
)
```
The K parameter is set in `KFoldSplitter` and then the splitter is used in `LinearModelValidation`. With this API, there is no need for a `ResourceMap` key, isn'it?

Reply:
My bad I had in mind things and wrote others.  

What I had in mind was more to avoid the creation (by users of) `KFoldSplitter` or `LeaveOneOutSplitter` and have something more generic like

```cxx
FunctionalChaosValidation(const FunctionalChaosResult & chaosResult,
                          const UnsignedInteger kFold = ResourceMap::GetAsUnsignedInteger("FunctionalChaosValidation-DefaultKFoldParameter");
```